### PR TITLE
core: fix wal saving an empty xlog during startup failure

### LINKED ISSUE #8704: Tarantool leaves `00...00.xlog` on exit during first box.cfg regardless of actual vclock
**Bug description**

Stopping Tarantool during initial `box.cfg()` (no matter, local recovery or bootstrap) makes it produce an empty `00000000000000000000.xlog` file.

<details><summary>Steps to reproduce</summary>

Example 1 (with a single instance):
```
./tarantool -e "box.cfg{} os.exit()"
```
```
./tarantool
> tarantool> box.cfg{replication='1234'}
-- While the node hangs, stop it with Ctrl+C and observe:
2023-05-25 11:27:12.202 [44406] wal/101/main xlog.c:856 E> SystemError: file './00000000000000000000.xlog' already exists: Fil 
```

Example 2 (with 2 nodes):
```lua
--
-- Instance 1.
--

-- Step 1.
box.cfg{listen = 3301}

-- Step 5.
box.schema.user.grant('guest', 'replication')

--
-- Instance 2.
--

-- Step 2.
box.cfg{replication = 3301}

-- Step 3.
Ctrl + C, then restart the instance

-- Step 4.
box.cfg{replication = 3301}

-- Result:
2023-05-25 12:35:53.257 [24789] main/103/interactive xlog.c:462 E> XlogError: ./00000000000000000000.xlog: invalid instance UUID
```
</details>

Reproduced on:
Tarantool 3.0.0-entrypoint-261-gcde911d0e
Tarantool 2.11.0-0-g247a9a418

Tarantool 2.10.7-0-g60f7e1858 - no issues, works just fine.

Bisect shows that commit a392eb765c12d5aa0d5b2ef538c94f46915e3f3e is the reason. This is quite logical: that commit made tarantool do box_free() even if box isn't configured yet. So the issue with wal_free() creating an empty xlog unconditionally was uncovered. Need to fix wal_free().

### PR DESCRIPTION
**Pull request by sergepetrenko**
After the commit a392eb765c12 ("box: destroy its modules even if box.cfg not done") wal_free() on exit is called even if box.cfg() isn't finished yet, and thus wal isn't yet enabled. This leads to a bug when wal would unconditionally write a 00...0.xlog regardless of the actual xlog directory contents or the actual vclock.

Let's fix this by not writing an empty xlog on exit if wal writer's vclock is zero. It means the writer either wasn't enabled yet, or was enabled, but hasn't written anything. In both cases writing the empty xlog is extraneous.

Closes #8704

NO_DOC=bugfix

### CODE REVIEW

**Review by Gerold103 — APPROVED** [review]
Thanks for the patch!

File: changelogs/unreleased/gh-8704-wal-exit-empty-xlog.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## bugfix/core
+
+* Fixed a bug which made the node write an empty `00000000000000000000.xlog`
+  file regardless of actual vclock when interrupted during the initial
+  `box.cfg()` call (gh-8704).
```
Comment (xuniq) [inline changelogs/unreleased/gh-8704-wal-exit-empty-xlog.md:5]:
```suggestion
* Fixed the node writing an empty `00000000000000000000.xlog`
  file regardless of the actual vclock when interrupted during the initial
  `box.cfg()` call (gh-8704).
```

**Review by xuniq — APPROVED** [review]

**Review by grafin — APPROVED** [review]
Thanks for the patch!

**Comment by sergepetrenko** [comment]
Will cherry-pick to 2.11 in scope of https://github.com/tarantool/tarantool/pull/8780

**Pull request merged**